src/rpc/blockchain.cpp

```diff
@@ -2550,7 +2563,8 @@ static RPCHelpMan scanblocks()
 
         ret.pushKV("from_height", start_block_height);
         ret.pushKV("to_height", start_index->nHeight); // start_index is always the last scanned block here
-        ret.pushKV("relevant_blocks", std::move(blocks));
+        LOCK(cs_relevant_blocks);
+        ret.pushKV("relevant_blocks", g_relevant_blocks);
```

Comment:
Since status stops working when the scan completes, it seems like we should clear `g_relevant_blocks` here?

Reply:
Yes, thank you.  Will update.

Reply:
While leaving `scanblocks()` without resetting `g_relevant_blocks` isn't ideal, resetting `g_relevant_blocks` just before `BlockFiltersScanReserver` is used by the `start` branch seemed to be a good way to prevent `status` from accidentally seeing an empty `g_relevant_blocks` (at least in common cases). Please let me know if I'm overlooking a likely concurrency case, and it can be adjusted.

Reply:
Seems trivial to fix properly? https://github.com/luke-jr/bitcoin/commit/56981318da5dc3145e9f244ad0584ef05c815f6b

Reply:
included, thanks